pkg/sources/sources.go

```diff
@@ -37,7 +103,7 @@ type Source interface {
 	// JobID returns the initialized job ID used for tracking relationships in the DB.
 	JobID() int64
 	// Init initializes the source.
-	Init(aCtx context.Context, name string, jobId, sourceId int64, verify bool, connection *anypb.Any, concurrency int) error
+	Init(aCtx context.Context, cfg SourceConfig) error
```

Comment:
**Idea:** it would be nice to have the strictly required information be passed as arguments and the rest as variadic args.

I *think* `connection` is the only strictly required item.

```go
source := git.Source{}
source.Init(ctx, conn,
    sources.WithName("foo"),
    sources.WithJobID(123),
    sources.WithVerify(true),
)
```

Another idea is the same but using the config struct instead of variadic args:

```go
source.Init(ctx, conn, &sources.Config{
    Name:   "foo",
    JobID:  123,
    Verify: true,
})
```

Reply:
Or if not here, then on `NewSourceConfig`

```go
source.Init(ctx, sources.NewConfig(conn,
    sources.WithName("foo"),
    sources.WithJobID(123),
    sources.WithVerify(true),
))
```